internal/state/store.go

```diff
@@ -202,7 +208,13 @@ func (s *changeTrackingUpdater) Upsert(obj client.Object) {
 
 	relationshipExists := s.capturer.Exists(obj, client.ObjectKeyFromObject(obj))
 
-	s.changed = s.changed || changingUpsert || relationshipExisted || relationshipExists
+	forceChanged := s.forceChangedByObject(obj, client.ObjectKeyFromObject(obj))
+
+	// FIXME(pleshakov): Check generation in all cases to minimize the number of Graph regeneration.
+	// s.changed can be true even if the generation of the object did not change, because
+	// capturer and forceChangedByObject don't take the generation into account.
+
+	s.changed = s.changed || changingUpsert || relationshipExisted || relationshipExists || forceChanged
```

Comment:
Since Secrets are now a part of the `persistedGVKs` in the `changeTrackingUpdater`, won't `changingUpsert` always be true when a Secret is created or updated even if it's not referenced by a listener?

Same for delete.

Reply:
it should not because trackingUpsertDelete is false
```
			{
				gvk:               extractGVK(&apiv1.Secret{}),
				store:             newObjectStoreMapAdapter(clusterStore.Secrets),
				trackUpsertDelete: false,
			},
```

Reply:
Thanks, I missed that case.